laskyawm-clock.js

```diff
@@ -0,0 +1,21 @@
+/* This file is licensed under the Affero General Public License. */
+
+(function(){
+	// Create clock widget
+	var clock = document.createElement('span');
+	clock.classList.add('barButton');
+	clock.id = "bar-clock";
+	updateTime();
+	
+	// Add to bar
+	bar.addItem(clock);
+
+	// Update clock on time changes (500 to account for errant synchronization)
+	setInterval(updateTime,500);
```

Comment:
Could you line this up entirely by doing a `setTimeout` with `1000 - currentTime.getMilliseconds()` after you update the time?